vm/state_manager/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+#![allow(dead_code)]
+
+mod errors;
+
+pub use self::errors::*;
+use actor::{MinerInfo, StorageMinerActorState};
+use address::Address;
+use amt::BlockStore;
+use chain::ChainStore;
+use encoding::de::DeserializeOwned;
+use state_tree::{HamtStateTree, StateTree};
+
+/// Intermediary for retrieving state objects and updating actor states
+pub struct StateManager<'a> {
+    cs: &'a ChainStore<'a>,
+    tree: HamtStateTree,
+}
+
+impl<'a> StateManager<'a> {
+    /// constructor
+    pub fn new(cs: &'a ChainStore, tree: HamtStateTree) -> Self {
+        Self { cs, tree }
+    }
+    /// Loads actor state from IPLD Store
+    fn load_actor_state<T>(&self, addr: &Address) -> Result<T, Error>
```

Comment:
Why is the actor state generic type? Won't this always be the same struct?

Reply:
No, in its current usage it could be the `StorageMinerActorState ` or `MinerInfo` struct